packages/config/src/project/index.ts

```diff
@@ -105,9 +105,17 @@ export function updateWithPluginValues (cfg: FullConfig, modifiedConfig: any, te
     debug('resolved config object %o', cfg.resolved)
   }
 
+  const diffsClone = _.cloneDeep(diffs)
+
   // merge cfg into overrides
   const merged = _.defaultsDeep(diffs, cfg)
 
+  for (const key in diffsClone) {
+    if (Array.isArray(diffsClone[key])) {
+      merged[key] = _.cloneDeep(diffsClone[key])
+    }
+  }
+
```

Comment:
```suggestion
  for (const [key, value] of Object.entries(diffsClone)) {
    if (Array.isArray(value)) {
      merged[key] = _.cloneDeep(value)
    }
  }
```

I think this can be a bit more concise. I did not try this, but it should work. 

I've also added a test. [47c12f1e4e712f62b0cec582b7dfda98f3177583](https://github.com/cypress-io/cypress/pull/27240/commits/47c12f1e4e712f62b0cec582b7dfda98f3177583)

You could use this to see if my refactor works. `cd packages/app` then `yarn dev`. Select the spec (or use `it.only` to only run the one test I added).

Reply:
Hello @lmiller1990. I made the suggested changes and locally tried the original reproduction repo, and it works perfectly fine for it. However, I was not able to run the tests locally. After selecting the spec, I was met with a lot of errors. I modified the `it()` block under which you had added these tests to `it.only()`, however it gave me the following error.

```
cy.task('__internal_withCtx') timed out after waiting 60000ms.



Because this error occurred during a before each hook we are skipping the remaining tests in the current suite: specChange subscription
```